Tests/SentryTests/Integrations/Performance/Network/SentryNetworkTrackerTests.swift

```diff
@@ -559,6 +509,72 @@ class SentryNetworkTrackerTests: XCTestCase {
         //Test if it has observers. Nil means no observers
         XCTAssertNil(task.observationInfo)
     }
+
+    func testBaggageHeader() {
+        let sut = fixture.getSut()
+        let task = createDataTask()
+        let transaction = startTransaction() as! SentryTracer
+        sut.urlSessionTaskResume(task)
+
+        let expectedBaggageHeader = transaction.traceContext.toBaggage().toHTTPHeader()
+        XCTAssertEqual(task.currentRequest?.allHTTPHeaderFields?["baggage"] ?? "", expectedBaggageHeader)
+    }
+
+    func testTraceHeader() {
+        let sut = fixture.getSut()
+        let task = createDataTask()
+        let transaction = startTransaction() as! SentryTracer
+        sut.urlSessionTaskResume(task)
+
+        let children = Dynamic(transaction).children as [SentrySpan]?
+        let networkSpan = children![0]
+        let expectedTraceHeader = networkSpan.toTraceHeader().value()
+        XCTAssertEqual(task.currentRequest?.allHTTPHeaderFields?["sentry-trace"] ?? "", expectedTraceHeader)
+    }
+
+    func testNoHeadersForWrongUrl() {
+        fixture.options.tracePropagationTargets = ["www.example.com"]
+
+        let sut = fixture.getSut()
+        let task = createDataTask()
+        _ = startTransaction() as! SentryTracer
+        sut.urlSessionTaskResume(task)
+
+        XCTAssertEqual(task.currentRequest?.allHTTPHeaderFields?["baggage"] ?? "", "")
+        XCTAssertEqual(task.currentRequest?.allHTTPHeaderFields?["sentry-trace"] ?? "", "")
+    }
+
+    func testAddHeadersForRequestWithURL() {
```

Comment:
`l`: Excellent test case, but I think we should split it up into several ones.

Reply:
Just wondering why you think so? Splitting it up doesn't really do anything, and I like how all these related things are now in one test function.